proxy/apifilterproxy/filters/auth/rancher_token_validate_filter.go

```diff
@@ -0,0 +1,289 @@
+package auth
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"strings"
+
+	log "github.com/Sirupsen/logrus"
+
+	"github.com/rancher/websocket-proxy/proxy/apifilterproxy/filters"
+	"github.com/rancher/websocket-proxy/proxy/apifilterproxy/model"
+)
+
+const (
+	name = "authTokenValidator"
+)
+
+//AuthorizeData is for the JSON output
+type AuthorizeData struct {
+	Message string `json:"message,omitempty"`
+}
+
+//MessageData is for the JSON output
+type MessageData struct {
+	Data []interface{} `json:"data,omitempty"`
+}
+
+func init() {
```

Comment:
I also think you can combine the logic in the manager package with the logic in the filters/api_filter.go and apfilterproxy/handler.go

You should be able to achieve one cohesive struct that encapsulates all this logic. The manager package should not have to expose any package level variables publicly.